src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs

```diff
@@ -329,41 +326,99 @@ static async Task<HttpResponseMessage> Impl(HttpRequestMessage request, Cancella
         }
     }
 
-    internal sealed class ReadableStreamPullState
+    internal sealed class WasmHttpWriteStream : Stream
     {
-        private readonly Stream _stream;
-        private readonly CancellationToken _cancellationToken;
-        private readonly byte[] _buffer;
+        private readonly JSObject _transformStream;
 
-        public ReadableStreamPullState(Stream stream, CancellationToken cancellationToken)
+        public WasmHttpWriteStream(JSObject transformStream)
         {
-            ArgumentNullException.ThrowIfNull(stream);
+            ArgumentNullException.ThrowIfNull(transformStream);
 
-            _stream = stream;
-            _cancellationToken = cancellationToken;
-            _buffer = new byte[65536];
+            _transformStream = transformStream;
         }
 
-        public async Task PullAsync()
+        public static async Task<JSObject> CopyToAsync(Task<JSObject> fetchResponsePromise, JSObject transformStream, HttpContent content, CancellationToken cancellationToken)
         {
-            try
-            {
-                int length = await _stream.ReadAsync(_buffer, _cancellationToken).ConfigureAwait(true);
-                ReadableStreamControllerEnqueueUnsafe(this, _buffer, length);
-            }
-            catch (Exception ex)
+            using (WasmHttpWriteStream stream = new WasmHttpWriteStream(transformStream))
             {
-                BrowserHttpInterop.ReadableStreamControllerError(this, ex);
+                try
+                {
+                    await content.CopyToAsync(stream, cancellationToken).ConfigureAwait(true);
+                    await BrowserHttpInterop.TransformStreamClose(transformStream).ConfigureAwait(true);
+                    return await fetchResponsePromise.ConfigureAwait(true);
```

Comment:
could this await move back up to CallFetch ?

Reply:
Yes, I have rearranged the code.